src/firefly/js/charts/ui/ChartsContainer.jsx

```diff
@@ -4,92 +4,40 @@
 
-import {ChartsTableViewPanel} from './ChartsTableViewPanel.jsx';
-import {getExpandedChartProps, getChartIdsInGroup} from '../ChartsCntlr.js';
-import {getActiveTableId} from '../../tables/TableUtil.js';
-
-
-function nextState(props) {
-
-    const {closeable, chartId} = props;
-    let {tblId} = props;
-    tblId = (tblId || chartId) ? tblId : getActiveTableId();
-
-    const currentCharts = chartId ? [chartId] : getChartIdsInGroup(tblId);
-    const defaultCharts = getChartIdsInGroup('default');
-
-    const expandedMode = props.expandedMode && getExpandedMode() === LO_VIEW.xyPlots;
-    const chartProps = expandedMode ? getExpandedChartProps() : {};
-
-    return Object.assign({expandedMode, closeable, currentCharts, defaultCharts}, chartProps);
-}
 
+/**
+ * Default viewer
+ */
 export class ChartsContainer extends Component {
     constructor(props) {
         super(props);
-        this.state = nextState(this.props);
-    }
-
-    componentWillReceiveProps(np) {
-        if (this.iAmMounted && !shallowequal(this.props, np)) {
-            this.setState(nextState(np));
-        }
-    }
-
-    componentDidMount() {
-        this.removeListener= flux.addListener(() => this.storeUpdate());
-        this.iAmMounted = true;
-    }
 
-    componentWillUnmount() {
-        this.iAmMounted = false;
-        this.removeListener && this.removeListener();
     }
 
     shouldComponentUpdate(nProps, nState) {
         return sCompare(this, nProps, nState);
     }
 
-
-    storeUpdate() {
-        if (this.iAmMounted) {
-            const expandedMode = this.props.expandedMode && getExpandedMode() === LO_VIEW.xyPlots;
-            if (expandedMode !== this.state.expandedMode || !(this.props.chartId||this.props.tblId)) {
-                const ns = nextState(this.props);
-                if (!isEqual(this.state.currentCharts, ns.currentCharts) ||
-                    !isEqual(this.state.defaultCharts, ns.defaultCharts) ) {
-                    this.setState(nextState(this.props));
-                }
-            }
-        }
-    }
-
     render() {
-        const {currentCharts, defaultCharts, expandedMode, closeable} = this.state;
-        return expandedMode ? <ExpandedView key='chart-expanded' closeable={closeable} {...this.props} {...this.state}/> :
-            (
-                <div style={{ display: 'flex',  height: '100%', flexGrow: 1, flexDirection: 'row', overflow: 'hidden'}}>
-                    {currentCharts.map((c) => {
-                        return (
-                            <ChartsTableViewPanel key={c} expandable={true} chartId={c}/>
-                        );
-                    })}
-                    {defaultCharts.map((c) => {
-                        return (
-                            <ChartsTableViewPanel key={'default-'+c} expandable={true} chartId={c}/>
-                        );
-                    })}
-                </div>
+        const {chartId, expandedMode, closeable} = this.props;
```

Comment:
This component handles single chart display from API (both standard and expanded mode) and wraps the default MultiChartViewer. Perhaps, at some point it can be eliminated, but I like to have this wrapper so that only chart code changes when chart code is refactored.